bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -308,6 +333,34 @@ where
 			[GlobalConsensus(network), AccountKey20 { network: None, key: token.into() }],
 		)
 	}
+
+	fn convert_transact(
+		message_id: H256,
+		chain_id: u64,
+		sender: H160,
+		origin_kind: OriginKind,
+		fee: u128,
+		weight_at_most: Weight,
+		payload: Vec<u8>,
+	) -> (Xcm<()>, Balance) {
+		// Fee in native token of destination chain
+		let xcm_fee: Asset = (Location::here(), fee).into();
+
+		let message = vec![
+			DescendOrigin(PalletInstance(InboundQueuePalletInstance::get()).into()),
+			// Change origin to the bridge.
+			UniversalOrigin(GlobalConsensus(Ethereum { chain_id })),
+			// DescendOrigin to the sender.
+			DescendOrigin(AccountKey20 { network: None, key: sender.into() }.into()),
+			WithdrawAsset(xcm_fee.clone().into()),
+			// Pay for execution.
+			BuyExecution { fees: xcm_fee, weight_limit: Unlimited },
+			Transact { origin_kind, require_weight_at_most: weight_at_most, call: payload.into() },
+			// Forward message id
+			SetTopic(message_id.into()),
+		];
+		(message.into(), fee.into())
```

Comment:
I would return `0` as a fee here. The returned fee is meant to be burnt.

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/114/commits/9829de7a0e12816c96c820b91abe195af8486269